### ISSUE TITLE
Proposal: auto-commit bot for allocating operator version numbers

### ISSUE THREAD

**Issue by ezyang** [onnx/onnx#332]
At the moment, operator set versions are a global monotonically increasing number which otherwise has no structure. The problem I have noticed is that people keep forgetting to update the version number when they make changes.

I propose that we create a commit bot to automatically allocate operator numbers for users. Here is how it will work:

1. Instead of setting `SinceVersion(3)` with a specific number, users will instead commit and push `SinceVersion(UNALLOCATED_VERSION)`. This sigil will be some variable whose value tells ONNX that the version number is unallocated
2. A bot monitors the repo for commits which reference `UNALLOCATED_VERSION` in the diff. When it detects such a commit, it generates and pushes a commit replacing this placeholder with a real version (also regenerating the docs)

What do people think?

CC @donbox

**Comment by dzhulgakov** [comment]
Maybe just have a contbuild thing that verifies that SinceVersion is specified everywhere?

**Comment by gramalingam** [comment]
@ezyang , do you mean that people remember to update "SinceVersion(...)" but forget to update the global version? (I first thought that you meant people forget to update "SinceVersion(...)", and wasn't clear how the proposal would fix that.)

Couldn't the bot also look for any "SinceVersion(..)" in the diff, and check if the global version is also in the diff?

**Comment by ezyang** [comment]
Yes, we could also just do that. In that case no committing isn't needed.

@dzhulgakov I've considered a contbuild thing; the difficult thing is that the test really needs to be done on a *diff*, you can't figure it out just from the state of the tree.
